tensorboard/plugins/hparams/backend_context.py

```diff
@@ -17,40 +17,39 @@
 from __future__ import division
 from __future__ import print_function
 
-import six
-
 from tensorboard.plugins.hparams import metadata
-from tensorboard.plugins.hparams import error
 
-class Context:
-  """Stores data shared across API handlers for the HParams plugin backend."""
-  def __init__(self, multiplexer):
+class Context(object):
+  """Extends the base_plugin.TBContext to stores additional data shared across
+  API handlers for the HParams plugin backend."""
+  def __init__(self, tb_context):
     """Instantiates a context.
     Args:
-      multiplexer: A plugin_event_multiplexer.EventMultiplexer instance
-        accessing the currently loaded runs and tags.
+      tb_context: base_plugin.TBContext. The "base" context we extend.
     """
+    self._tb_context = tb_context
     self._experiment = None
-    self._multiplexer = multiplexer
 
   def experiment(self):
-    # Note: We can't search for the experiment in the constructor,
-    # since this object may be initialized before Tensorboard reads
-    # the event data.
+    """Searches for the experiment tag and returns the associated experiment
+    protobuffer. If no tag is found (possibly, because the event data has not
+    been completely loaded yet), returns None.
+    """
     if self._experiment is None:
       self._experiment = self._find_experiment()
     return self._experiment
 
   def multiplexer(self):
-    return self._multiplexer
+    return self._tb_context.multiplexer
```

Comment:
Ok. But I'd like to make sure that Context.multiplexer can't be changed to reference another object.

Reply:
You can do this with `@property` by just not providing a setter, e.g. just decorate `multiplexer` with `@property` and it should work like it does now but be accessible as `.multiplexer`.  See e.g.  https://github.com/google/styleguide/blob/gh-pages/pyguide.md#2134-decision

Also FWIW, while I understand the instinct to code defensively, ultimately in Python there's very little true defensive coding ability that doesn't rest on convention to some degree; someone can just change `_tb_context` if they want to.  And even if you've defined `multiplexer` as a method, there's no reason why somebody couldn't take your Context object and just do:

```python
context = getContext()
context.multiplexer = lambda: 42
assert context.multiplexer() == 42
```

In that respect `@property` is actually more "read-only" since you wouldn't be able to do that, at least not as easily.

Reply:
Ok. Done that. Yep, someone can always do that. I'm trying to defend agains unintenional mistakes rather than intentionally breaking the rules :-), but I guess in the end it's a spectrum and I'm more on the defensive side.